### PR TITLE
GDPR element

### PR DESCRIPTION
**Pull request by esokoll2**
New form element to display a GDPR link next to a checkbox

### CODE REVIEW

File: Configuration/Form/Checkboxlink.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+TYPO3:
+  CMS:
+    Form:
+      prototypes:
+        standard:
+          formEditor:
+            dynamicRequireJsModules:
+              additionalViewModelModules:
+                - 'TYPO3/CMS/FormTools/Backend/FormEditor/CheckboxlinkViewmodel'
```
Comment (froemken) [inline Configuration/Form/Checkboxlink.yaml:9]: ViewModel. 'm' uppercase as you can see in line above

File: Documentation/Configuration/Index.rst [hunk 2]
```diff
@@ -37,4 +37,43 @@ of EXT:form
...
              value: 'Default Subject from YAML'
+           pid:
+             value: 1
+           tstamp:
+             value: '{__currentTimestamp}'
+           crdate:
+             value: '{__currentTimestamp}'
        identifier: StoreFieldsAsXmlToDb
+
+Set the value of pid to your page/folder where the mails should be found in the BackEnd.
```
Comment (froemken) [inline Documentation/Configuration/Index.rst:48]: Backend. lowercased e

File: Resources/Private/FormFramework/Standard/Frontend/Partials/Checkboxlink.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+                    {formvh:translateElementProperty(element: element, property: 'label')}
+                    -->
+                </f:comment>
+                <span>Ich habe die <f:link.page pageUid="{f:cObject(typoscriptObjectPath:'lib.formLegalUid')}" target="_blank">Datenschutzerklärung</f:link.page> gelesen und stimme dieser zu. <f:if condition="{element.required}"><f:render partial="Field/Required" /></f:if></span>
+            </label>
+        </div>
+    </f:render>
+</formvh:renderRenderable>
+</html>
```
Comment (froemken) [inline Resources/Private/FormFramework/Standard/Frontend/Partials/Checkboxlink.html:28]: Add new line behind

File: Resources/Private/FormFramework/Standard/Frontend/Partials/Checkboxlink.html [hunk 4]
```diff
@@ -0,0 +1,28 @@
+<html xmlns:f="http://typo3.org/ns/TYPO3/CMS/Fluid/ViewHelpers" xmlns:formvh="http://typo3.org/ns/TYPO3/CMS/Form/ViewHelpers" data-namespace-typo3-fluid="true">
```
Comment (froemken) [inline Resources/Private/FormFramework/Standard/Frontend/Partials/Checkboxlink.html:1]: Very long line. Please add some returns after each attribute like you have done with f:form.checkbox in line 6

File: Resources/Public/JavaScript/Backend/FormEditor/CheckboxlinkViewmodel.js [hunk 5]
```diff
@@ -0,0 +1,132 @@
+// Resources/Public/JavaScript/Backend/FormEditor/CheckboxlinkViewmodel.js
```
Comment (froemken) [inline Resources/Public/JavaScript/Backend/FormEditor/CheckboxlinkViewmodel.js:1]: upper case 'm'. This has to updated in filename itself, too.

File: Resources/Public/JavaScript/Backend/FormEditor/CheckboxlinkViewmodel.js [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+        /**
+         * Publish the public methods.
+         * Implements the "Revealing Module Pattern".
+         */
+        return {
+            bootstrap: bootstrap
+        };
+    })($, Helper);
+});
```
Comment (froemken) [inline Resources/Public/JavaScript/Backend/FormEditor/CheckboxlinkViewmodel.js:132]: Add new line behind

**Review by froemken — CHANGES_REQUESTED** [review]

File: Configuration/Form/Checkboxlink.yaml (hunk 1 above)
Comment (esokoll2) [reply Configuration/Form/Checkboxlink.yaml]: Done

File: Documentation/Configuration/Index.rst (hunk 2 above)
Comment (esokoll2) [reply Documentation/Configuration/Index.rst]: Done

File: Resources/Private/FormFramework/Standard/Frontend/Partials/Checkboxlink.html (hunk 3 above)
Comment (esokoll2) [reply Resources/Private/FormFramework/Standard/Frontend/Partials/Checkboxlink.html]: Done

File: Resources/Private/FormFramework/Standard/Frontend/Partials/Checkboxlink.html (hunk 4 above)
Comment (esokoll2) [reply Resources/Private/FormFramework/Standard/Frontend/Partials/Checkboxlink.html]: Done

File: Resources/Public/JavaScript/Backend/FormEditor/CheckboxlinkViewmodel.js (hunk 5 above)
Comment (esokoll2) [reply Resources/Public/JavaScript/Backend/FormEditor/CheckboxlinkViewmodel.js]: Done

File: Resources/Public/JavaScript/Backend/FormEditor/CheckboxlinkViewmodel.js (hunk 6 above)
Comment (esokoll2) [reply Resources/Public/JavaScript/Backend/FormEditor/CheckboxlinkViewmodel.js]: Done

**Review by froemken — APPROVED** [review]

**Pull request merged**